iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs

```diff
@@ -405,10 +406,6 @@ private async Task<Twin> RoundTripTwinMessage(AmqpTwinMessageType amqpTwinMessag
                     // Consider that the task may have faulted or been canceled.
                     // We re-await the task so that any exceptions/cancellation is rethrown.
                     response = await receivingTask.ConfigureAwait(false);
-                    if (response == null)
-                    {
-                        throw new InvalidOperationException("Service response is null");
-                    }
                 }
                 else
                 {
```

Comment:
How about we add some additional context to the TimeoutException being thrown below (and maybe a log statement as well)?